src/odemis/driver/test/orsay_test.py

```diff
@@ -0,0 +1,1037 @@
+        sleep(1)
+        self.assertEqual(self.gis_res.temperatureRegulation.value, 2)
+
+        self.gis_res._gis.RegulationOn.Target = False
+        self.gis_res._gis.RegulationRushOn.Target = True
+        sleep(1)
+        self.assertEqual(self.gis_res.temperatureRegulation.value, 2)
+
+        self.gis_res._gis.RegulationOn.Target = True
+        self.gis_res._gis.RegulationRushOn.Target = False
+        sleep(1)
+        self.assertEqual(self.gis_res.temperatureRegulation.value, 1)
+
+        self.gis_res._gis.RegulationOn.Target = False
+        self.gis_res._gis.RegulationRushOn.Target = False
+        sleep(1)
+        self.assertEqual(self.gis_res.temperatureRegulation.value, 0)
+
+    def test_updateAge(self):
+        """
+        Check that the age VA is updated correctly and an exception is raised when the wrong parameter is passed
+        """
+        with self.assertRaises(ValueError):
+            self.gis_res._updateAge(self.datamodel.HybridPlatform.Cancel)
+
+        if not TEST_NOHW == "sim":
+            self.skipTest("TEST_NOHW is not set to sim, cannot force data on Actual parameters of Orsay server "
+                          "outside of simulation.")
+
+        test_value = 20
+        self.gis_res._gis.ReservoirLifeTime.Actual = test_value
+        self.assertEqual(self.gis_res.age.value, test_value)
+        self.gis_res._gis.ReservoirLifeTime.Actual = 0
+        self.assertEqual(self.gis_res.age.value, 0)
+
+    def test_updatePrecursorType(self):
+        """
+        Check that the precursorType VA is updated correctly and an exception is raised when the wrong parameter is
+        passed
+        """
+        with self.assertRaises(ValueError):
+            self.gis_res._updatePrecursorType(self.datamodel.HybridPlatform.Cancel)
+
+        if not TEST_NOHW == "sim":
+            self.skipTest("TEST_NOHW is not set to sim, cannot force data on Actual parameters of Orsay server "
+                          "outside of simulation.")
+
+        test_value = "test precursor"
+        self.gis_res._gis.PrecursorType.Actual = test_value
+        self.assertEqual(self.gis_res.precursorType.value, test_value)
+        self.gis_res._gis.PrecursorType.Actual = "Simulation"
+        self.assertEqual(self.gis_res.precursorType.value, "Simulation")
+
+    def test_setTemperatureTarget(self):
+        """
+        Test the setter of the temperatureTarget VA
+        """
+        test_value = 20
+        self.gis_res.temperatureTarget.value = test_value
+        self.assertEqual(self.gis_res._temperaturePar.Target, test_value)
+        self.gis_res.temperatureTarget.value = 0
+        self.assertEqual(self.gis_res._temperaturePar.Target, 0)
+
+    def test_setTemperatureRegulation(self):
+        """
+        Test the setter of the temperatureRegulation VA
+        """
+        self.gis_res.temperatureRegulation.value = 0
+        sleep(0.5)
+        self.assertFalse(self.gis_res._gis.RegulationOn.Target.lower() == "true")
```

Comment:
After each sleep, please also check that temperatureRegulation is the right/same value.